Patch download and area calculation

sssangha opened the pull request:
A few bug-fixes were implemented:

1. In `ariaDownloads.py`, duplicate products (if they already existed) were originally not moved into separate subdirectory so they may not be passed along. This patch ensures such duplicate products are indeed segregated to this subdirectory.

2. Option added to `shapefile_area` function in `shapefile_util.py` to estimate the area of the region, as opposed to just the cutline, in order to obtain an accurate estimate of the extent of a downloaded DEM and/or mask. This resolves issue where DEMs weren’t chunked when they should have been, leading to crashes as the SRTM server cannot handle such large requests. Addresses #265 . 

3. Incremental updates to be inline with pep standards also passed.

sssangha committed:
Patch download and area calculation

sssangha committed:
shapefile_util codacy tweaks

sssangha committed:
extractproduct condacy fixes

dbekaert commented on tools/ARIAtools/extractProduct.py:
```diff
@@ -387,20 +378,24 @@ def prep_dem(demfilename, bbox_file, prods_TOTbbox, prods_TOTbbox_metadatalyr,
+    # If area > 450000 km2, must split requests into chunks to successfully access data
     chunk = False
-    if shapefile_area(prod_shapefile) > 300000:
+    if shapefile_area(prod_shapefile, bounds = True) > 450000:
```
@sssangha suggest to be conservative and make this sligthly smaller than the openTopo API.

dbekaert replied:
```suggestion
    if shapefile_area(prod_shapefile, bounds = True) > 400000:
```

dbekaert commented on tools/ARIAtools/extractProduct.py:
```diff
@@ -387,20 +378,24 @@ def prep_dem(demfilename, bbox_file, prods_TOTbbox, prods_TOTbbox_metadatalyr,
         # Increase chunking size to discretize box into smaller grids
-        log.warning('User-defined bounds results in an area of %d km which exceeds the maximum download area of 450000; downloading in chunks', shapefile_area(prod_shapefile))
+        log.warning('User-defined bounds results in an area of %d km which ' \
+                    'exceeds the maximum download area of 450000; ' \
```
```suggestion
                    'exceeds the maximum download area of 400000; ' \
```

dbekaert commented:
@rzinke @bbuzz31 Can you test this over your region to see if it remove the issue on the DEM download?

jlmaurer commented:
@sssangha I'm still having the same issue even after a clean start and pulling #275. DEM has all zeros and files that are not VRT files have no valid pixels (besides the unwrapped phase). This smells like it could be a GDAL issue, I'm using version:
`gdal                      3.1.0            py38h9f7df5a_3    conda-forge`
Could this be an issue?

jlmaurer commented:
Note that amplitude and coherence are VRTs and they work fine, have valid pixels etc.

bbuzzanga committed:
update NLCD to new LULC mask

bbuzzanga committed:
more conservative area calculation for dem chunk

bbuzzanga committed:
catch error when old files exists for different specified products

sssangha committed:
Capture duplicates with all executables

sssangha commented:
@dbekaert these updates now ensure that regardless of what executable the user launches (e.g. `ariaDownload.py`, `ariaExtract.py`, `ariaTSsetup.py`, and `ariaPlot.py`), duplicate products are captured and quarantined to a separate subfolder. I also cleaned up the code by moving this function out of `ariaDownload.py` and into a separate util script that is accessible to other scripts.

@bbuzz31 your contributions look good on my end! Was able to perform tests with a large DEM without a problem.

@jlmaurer I do recall issues with some relatively recent gdal update, but I was again unable to replicate your issue with `gdal 3.1.2`. Could you please retry again by updating to the specific version I vetted this code with? I.e.: `conda install gdal=3.1.2`.

If it is indeed an issue with older versions of gdal as I suspect, we may update the environment file accordingly.

sssangha committed:
GACOS update new format

bbuzzanga committed:
print error when caught

very simple update so that the error of why multiprocessing fails is printed to the screen

bbuzzanga committed:
fix logging bug

bbuzzanga reviewed (approved):
GACOS update LGTM
